tests/analysis/impl/dense/gpcca_test.py

```diff
@@ -0,0 +1,468 @@
+
+    def test_schur_b_pos(self):
+        N = 9
+        mu0 = mu(0)
+        P, sd = get_known_input(mu0)
+        X, RR, _ = _do_schur(P, eta=sd, m=3)
+
+        np.testing.assert_array_equal(P.shape, [N, N])
+        np.testing.assert_array_equal(X.shape, [9, 3])
+        np.testing.assert_array_equal(RR.shape, [3, 3])
+
+        assert np.all(np.abs(X @ RR - P @ X) < eps)
+        assert np.all(np.abs(X[:, 0] - 1) < eps)
+
+    def test_schur_b_neg(self):
+        mu0 = mu(0)
+        P, sd = get_known_input(mu0)
+        with pytest.raises(
+            ValueError,
+            match="The number of clusters/states is not supposed to be negative",
+        ):
+            _do_schur(P, eta=sd, m=-3)
+
+    def test_fill_matrix_not_square(self):
+        with pytest.raises(ValueError, match="Rotation matrix isn't quadratic."):
+            _fill_matrix(np.zeros((3, 4)), np.empty((3, 4)))
+
+    def test_fill_matrix_shape_error(self):
+        with pytest.raises(
+            ValueError,
+            match="The dimensions of the rotation matrix don't match with the number of Schur vectors",
+        ):
+            _fill_matrix(np.zeros((3, 3)), np.empty((3, 4)))
+
+    def test_gram_schmidt_shape_error_1(self):
+        with pytest.raises(ValueError):
+            _gram_schmidt_mod(np.array([3, 1]), np.array([1]))
+
+    def test_gram_schmidt_shape_error_2(self):
+        with pytest.raises(ValueError):
+            _gram_schmidt_mod(
+                np.array([3, 1]),
+                np.array(
+                    [np.true_divide(9, np.sqrt(10)), np.true_divide(1, np.sqrt(10))]
+                ),
+            )
+
+    def test_gram_schmidt_mod_R2(self):
+        Q = _gram_schmidt_mod(
+            np.array([[3, 1], [2, 2]], dtype=np.float64), np.array([0.5, 0.5])
+        )
+        s = np.sqrt(0.5)
+
+        orthosys = np.array([[s, -s], [s, s]])
+
+        assert_allclose(Q, orthosys)
+
+    def test_gram_schmidt_mod_R4(self):
+        Q = _gram_schmidt_mod(
+            np.array([[1, 1, 1, 1], [-1, 4, 4, 1], [4, -2, 2, 0]], dtype=np.float64).T,
+            np.array([0.25, 0.25, 0.25, 0.25]),
+        )
+        d = np.true_divide
+        s2 = np.sqrt(2)
+        s3 = np.sqrt(3)
+
+        u1 = np.array([0.5] * 4)
+        u2 = np.array([d(-1, s2), d(s2, 3), d(s2, 3), d(-1, 3 * s2)])
+        u3 = np.array([d(1, 2 * s3), d(-5, 6 * s3), d(7, 6 * s3), d(-5, 6 * s3)])
+        orthosys = np.array([u1, u2, u3]).T
+
+        assert_allclose(Q, orthosys)
+
+    def test_indexshape_shape_error(self):
+        with pytest.raises(ValueError):
+            _indexsearch(np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0]]))
+
+    @pytest.mark.xfail(reason="1st vector is not constant")
+    def test_indexsearch_1(self):
+        v = np.eye(6)
+        v = np.r_[np.zeros((1, 6)), v]
+
+        sys = np.c_[
+            v[0], v[1], v[0], v[2], v[0], v[3], v[0], v[4], v[0], v[5], v[0], v[6]
+        ].T
+
+        index = _indexsearch(sys)
```

Comment:
This fails because the `_indexsearch` method requires the first vector to be constant, which is not the case in this example here. 

@msmdev , your input would be appreciated here.

Reply:
Maybe you could modify the matlab example so that the first vector is constant, re-run the code in matlab and update the ground truth result?